packages/provider-elasticsearch/src/utils/futil.js

```diff
@@ -99,3 +99,21 @@ export let renameOn = (from, to, obj) => {
 // Async version of compactMap (and indexed)
 export let compactMapAsync = async (...args) =>
   _.compact(await Promise.all(F.mapIndexed(...args)))
+
+// Similar to F.getOrReturn but for _.set
+export const setOrReturn = (path, val, obj) =>
+  path ? _.set(path, val, obj) : val
+
+// _.groupBy but also passing the current key
+export const groupByIndexed = _.curry((it, coll) =>
```

Comment:
We should be able to call convert on lodash instead of reimplementing `groupBy`. I think we also expose `F.indexed` so you could do `F.indexed.groupBy`

Reply:
The normal (non-fp) `groupBy` does not pass the index

Reply:
Yup, that's right. We've done this in other places by composing `_.toPairs` with `_.groupBy`

Reply:
There would still be this util though but with a different implementation because it's not just `_.toPairs` and `_.groupBy`, after grouping there's `_.mapValues` to get rid of the pairs. And then you have an implementation that is probably as verbose as this one but less efficient.

Reply:
Do you have a better idea?